lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -369,25 +369,41 @@ namespace Js
         // Need to enumerate a different type than the current one. This is because type snapshot enumerate is enabled and the
         // object's type changed since enumeration began, so need to enumerate properties of the initial type.
         DynamicTypeHandler *const typeHandlerToEnumerate = typeToEnumerate->GetTypeHandler();
-
-        if (!typeHandlerToEnumerate->IsPathTypeHandler())
-        {
-            return typeHandlerToEnumerate->FindNextProperty(scriptContext, index, propertyStringName, propertyId, attributes, type, typeToEnumerate, flags, instance, info);
-        }
-
-        PathTypeHandlerBase* pathTypeToEnumerate = (PathTypeHandlerBase*)typeHandlerToEnumerate;
-
-        BOOL found = pathTypeToEnumerate->FindNextProperty(scriptContext, index, propertyStringName, propertyId, attributes, typeToEnumerate, typeToEnumerate, flags, instance, info);
-
-        // We got a property from previous type, but this property may have been deleted
-        if (found == TRUE && this->GetPropertyIndex(*propertyId) == Js::Constants::NoSlot)
-        {
-            PropertyValueInfo::SetNoCache(info, instance);
-            return FALSE;
+        for(
+            ;
+            typeHandlerToEnumerate->FindNextProperty(
+                scriptContext,
+                index,
+                propertyStringName,
+                propertyId,
+                attributes,
+                typeToEnumerate,
+                typeToEnumerate,
+                flags,
+                instance,
+                info);
+            ++index)
+        {
+            PropertyIndex tmpIndex = this->GetPropertyIndex(*propertyId);
+            if (tmpIndex != Constants::NoSlot)
+            {
+                ObjectSlotAttributes attr = objectAttrs ? objectAttrs[tmpIndex] : ObjectSlotAttr_Default;
+                if (!(attr & ObjectSlotAttr_Deleted) && attr != ObjectSlotAttr_Setter &&
+                    !!(flags & EnumeratorFlags::EnumNonEnumerable) || (attr & ObjectSlotAttr_Enumerable))
```

Comment:
Are we missing some parentheses here? It looks like we want to skip `ObjectSlotAttr_Setter`, but it would test as true for `(attr & ObjectSlotAttr_Enumerable)`.

Reply:
I just noticed this line is also causing the compilation failure on xplat; it doesn't like expressions using both `&&` and `||`, for good reason.

Reply:
Yeah, I think you're right about the parens.

Reply:
`&&` binds more tightly than `||` so this would combine all the ANDs before doing the OR.  Don't know whether that's correct but yeah, mixing AND and OR in the same expression without parens is asking for trouble.  clang will throw a fit if it sees such an expression.